### PR TITLE
Document argscheck

### PR DESCRIPTION
**Pull request by raphinesse**
Described behavior extracted from the current code. I am certain that I got it all covered.

### CODE REVIEW

**Comment by raphinesse** [comment]
TODO:
- Add the fact that `args` can also be an `arguments` object
- If `opt_callee` is missing, `args.callee` will be used
- Add example

Example:
```js
function foo (a, b, c) {
    checkArgs('sfO', 'my.package.foo', arguments);
    // ...
}
```

File: src/common/argscheck.js [hunk 1]
```diff
@@ -36,6 +36,35 @@ function extractParamName (callee, argIndex) {
...
+ *
+ * Lowercase characters specify arguments that must not be `null` or `undefined`
+ * while uppercase characters allow those values to be passed.
+ *
+ * Finally, `*` can be used to allow any type at the corresponding position.
+ *
+ * @example
+ * function foo (arr, opts) {
+ *     // require `arr` to be an Array and `opts` an Object, null or undefined
+ *     checkArgs('aO', 'my.package.foo', arguments);
```
Comment (janpio) [inline src/common/argscheck.js]:
```suggestion
 *     checkArgs('aO', 'my.package.foo', opts);
```
I guess? 
If not, and `arguments` would be created inside `foo` somewhere first, then maybe add `...` do indicate left out code.

File: src/common/argscheck.js [hunk 2]
```diff
@@ -36,6 +36,35 @@ function extractParamName (callee, argIndex) {
     return (/\(\s*([^)]*?)\s*\)/).exec(callee)[1].split(/\s*,\s*/)[argIndex];
 }
 
+/**
+ * Throws TypeError if the provided arguments do not have the expected types.
```
Comment (janpio) [inline src/common/argscheck.js]:
```suggestion
 * checkArgs can be used to check the type of method parameters
 *
 * Throws TypeError if the provided arguments do not have the expected types.
```

Comment (raphinesse) [reply src/common/argscheck.js]: I like mine better 😅

File: src/common/argscheck.js (hunk 1 above)
Comment (raphinesse) [reply src/common/argscheck.js]: No, the example is correct. [`arguments` is special](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments)

Comment (janpio) [reply src/common/argscheck.js]: TIL

File: src/common/argscheck.js (hunk 2 above)
Comment (janpio) [reply src/common/argscheck.js]: Mine is additional in front, as an introduction - felt that was missing.

Comment (raphinesse) [reply src/common/argscheck.js]: I'll try to incorporate it when I find some time

**Pull request merged**